packages/wrangler/src/pages/functions/routing-rules-util.test.ts

```diff
@@ -0,0 +1,117 @@
+import {
+	isRoutingRuleMatch,
+	transformRoutingRuleToRegExp,
+} from "./routing-rules-util";
+
+describe("isRoutingRuleMatch", () => {
+	it("should match rules referencing root level correctly", () => {
+		const routingRule = "/";
+
+		expect(isRoutingRuleMatch("/", routingRule)).toBeTruthy();
+		expect(isRoutingRuleMatch("/foo", routingRule)).toBeFalsy();
+		expect(isRoutingRuleMatch("/foo/bar", routingRule)).toBeFalsy();
+	});
+
+	it("should match include-all rules correctly", () => {
+		const routingRule = "/*";
+
+		expect(isRoutingRuleMatch("/", routingRule)).toBeTruthy();
+		expect(isRoutingRuleMatch("/foo", routingRule)).toBeTruthy();
+		expect(isRoutingRuleMatch("/foo/bar", routingRule)).toBeTruthy();
+		expect(isRoutingRuleMatch("/foo/bar/baz", routingRule)).toBeTruthy();
+	});
+
+	it("should match `/*` suffix-ed rules correctly", () => {
+		let routingRule = "/foo/*";
+
+		expect(isRoutingRuleMatch("/foo", routingRule)).toBeTruthy();
+		expect(isRoutingRuleMatch("/foo/", routingRule)).toBeTruthy();
+		expect(isRoutingRuleMatch("/foobar", routingRule)).toBeFalsy();
+		expect(isRoutingRuleMatch("/foo/bar", routingRule)).toBeTruthy();
+		expect(isRoutingRuleMatch("/foo/bar/baz", routingRule)).toBeTruthy();
+		expect(isRoutingRuleMatch("/bar/foo", routingRule)).toBeFalsy();
+		expect(isRoutingRuleMatch("/bar/foo/baz", routingRule)).toBeFalsy();
+
+		routingRule = "/foo/bar/*";
+
+		expect(isRoutingRuleMatch("/foo", routingRule)).toBeFalsy();
+		expect(isRoutingRuleMatch("/foo/", routingRule)).toBeFalsy();
+		expect(isRoutingRuleMatch("/foo/bar", routingRule)).toBeTruthy();
+		expect(isRoutingRuleMatch("/foo/bar/baz", routingRule)).toBeTruthy();
+		expect(isRoutingRuleMatch("/foo/barfoo", routingRule)).toBeFalsy();
+		expect(isRoutingRuleMatch("baz/foo/bar", routingRule)).toBeFalsy();
+		expect(isRoutingRuleMatch("baz/foo/bar/", routingRule)).toBeFalsy();
+	});
+
+	it("should match rules without wildcards correctly", () => {
+		let routingRule = "/foo";
+
+		expect(isRoutingRuleMatch("/foo", routingRule)).toBeTruthy();
```

Comment:
Need a couple more tests here:
- `/foo` should match `/foo/`
- `/foo/` should match `/foo`